wandb/sdk/launch/runner/local.py

```diff
@@ -75,23 +77,18 @@ def run(self, launch_project: LaunchProject) -> Optional[AbstractRun]:
         validate_docker_installation()
         synchronous: bool = self.backend_config[PROJECT_SYNCHRONOUS]
         docker_args: Dict[str, Any] = self.backend_config[PROJECT_DOCKER_ARGS]
-        entry_point = launch_project.get_single_entry_point()
 
         if launch_project.docker_image:
             # user has provided their own docker image
             _logger.info("Pulling user provided docker image")
-            pull_docker_image(launch_project.docker_image)
-
-            wandb.termwarn(
-                "Using supplied docker image: {}. Artifact swapping and launch metadata disabled".format(
-                    launch_project.docker_image
-                )
-            )
+            if not docker_image_exists(launch_project.docker_image):
+                pull_docker_image(launch_project.docker_image)
```

Comment:
nice catch

Reply:
ah, this isn't necessary anymore, we do the check in pull_docker_image